### PR TITLE
Add Docs For Prometheus Metrics

### PR DESCRIPTION
**Pull request by seanmalloy**
Prometheus metrics were added as part of the Vault v1.1.0 release in PR #5308. But no documentation was created. Adds the telemetry configuration docs and the API docs.

### CODE REVIEW

**Comment by riuvshin** [comment]
It seems that API Docs also should be updated: https://www.vaultproject.io/api/system/

**Comment by kalafut** [comment]
@seanmalloy re: the OpenAPI output, query parameters in general aren't being handled correctly. I'll look into fixing that.

**Comment by seanmalloy** [comment]
I added the missing API docs for /v1/sys/metrics. Let  me know if any changes are needed.

File: website/source/api/system/metrics.html.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+## Read Telemetry Metrics
+
+This endpoint returns the telemetry metrics for Vault. It can be used by metrics
+collections systems like [Prometheus](https://prometheus.io) that use a pull
+model for metrics collection.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `GET`    | `/sys/metrics`               | `000 application/json` |
```
Comment (kalafut) [inline website/source/api/system/metrics.html.md]: The "Produces" column can be removed. We just today made this change broadly (https://github.com/hashicorp/vault/commit/d096f62d8dd49836bf0500bd33942f7dde85f5df).

File: website/source/docs/configuration/telemetry.html.md [hunk 2]
```diff
@@ -135,3 +135,31 @@ These `telemetry` parameters apply to
...
+### `prometheus`
+
+These `telemetry` parameters apply to
+[prometheus](https://prometheus.io).
+
+* `prometheus_retention_time` `(string: "24h")` - Specifies the amount of time that
+  prometheus metrics are retained in memory. It is recommended to also enable the option
+  `disable_hostname` to avoid having prefixed metrics with hostname. Vault does not use the
+  default Prometheus path, so Prometheus must be configured as follows. Note that using
+  ?format=prometheus in the path won't work as ? will be escaped, so it must be specified
```
Comment (kalafut) [inline website/source/docs/configuration/telemetry.html.md]:
```suggestion
  `?format=prometheus` in the path won't work as "?" will be escaped, so it must be specified
```

File: website/source/docs/configuration/telemetry.html.md [hunk 3]
```diff
@@ -135,3 +135,31 @@ These `telemetry` parameters apply to
...
+  as a parameter. Addtionally a Vault token is required to access /v1/sys/metrics. The prometheus
```
Comment (kalafut) [inline website/source/docs/configuration/telemetry.html.md]:
```suggestion
  as a parameter. Additionally a Vault token is required to access /v1/sys/metrics. The prometheus
```

**Comment by kalafut** [comment]
This looks good! Thanks for putting it together. I left a few inline comments. In addition, since the metrics page is new, there is another change needed in order for it to show up in the sidebar. Please add `metrics` before `mounts` here:  https://github.com/hashicorp/vault/blob/3af9be469150b76aa035545f522a7efbf36ffc6d/website/source/layouts/api.erb#L117

I'll handle the OpenAPI updates separately.

File: website/source/api/system/metrics.html.md (hunk 1 above)
Comment (seanmalloy) [reply website/source/api/system/metrics.html.md]: Fixed

**Comment by seanmalloy** [comment]
@kalafut thanks for the review. I have addressed all of your review comments. Let me know if you want me to clean up the commit history before merging.

**Comment by kalafut** [comment]
@seanmalloy Looks good, thanks again!

**Comment by kalafut** [comment]
OpenAPI updates:  https://github.com/hashicorp/vault/pull/6490

**Pull request merged**